criblio/appscope
# scope stop

**This PR**
Adds the `scope stop` command to the CLI to allow users to effectively undo most of the effects of the `scope start` command.

**Automated Testing**
A `scope stop` integration test has been added to verify that:
- Data stops flowing (scope is in detached state)
- Service configurations are removed
- /etc/profile.d/scope.sh has been removed
- The filter file has been removed

**QA Instructions**
```
For these instructions please check all relevant paths (/tmp and /usr/lib)
For these instructions please also test the commands in a container and on the host
- Use scope start with a filter file to start scoping a filtered list of processes and services
- Verify that the processes are scoped ; service configs are in place ; profile.d/scope.sh exists; filter file exists
- Use scope stop to stop scoping all processes and services (and prevent future apps from being scoped automatically)
- Verify that data stops flowing ; service configs are removed ; profile.d/scope.sh is removed; filter file is removed
```

**Intended Usage**
Users should be able to use the feature like so:
```
docker run --rm -it -v /:/hostfs:ro --privileged cribl/scope:next
scope stop
```

Closes: #1045
